### PR TITLE
configstate: simplify ConfigManager

### PR DESCRIPTION
**Pull request by mvo5**
The ConfigManager is no longer running tasks on its own. All the
heavy lifting is done by the HookManager. So the TaskRunner and
State that ConfigManager used to have is now obsolete and can
be removed.

We could clean this up further by not even registering the ConfigManager in the overlord. Please let me know if you prefer that.

### CODE REVIEW

**Comment by pedronis** [comment]
before it was the odd one out, I think either we keep this way, or Manager needs a different name and not return a struct anymore (none of this is urgent)

File: overlord/configstate/configmgr.go [hunk 1]
```diff
@@ -44,8 +36,7 @@ func MockCorecfgRun(f func(conf corecfg.Conf) error) (restore func()) {
 	}
 }
 
-// Manager returns a new ConfigManager.
-func Manager(st *state.State, hookManager *hookstate.HookManager) (*ConfigManager, error) {
+func Init(hookManager *hookstate.HookManager) error {
```
Comment (pedronis) [inline overlord/configstate/configmgr.go]: do we need the error return?

File: overlord/overlord.go [hunk 2]
```diff
@@ -139,6 +134,10 @@ func New() (*Overlord, error) {
 
 	o.addManager(cmdstate.Manager(s))
 
+	if err := configstate.Init(hookMgr); err != nil {
```
Comment (pedronis) [inline overlord/overlord.go]: if I understand the intention, this should be configstateInit ?

Comment (mvo5) [reply overlord/overlord.go]: Heh, indeed, silly me. Fixed.

**Comment by bboozzoo** [comment]
Travis build hit a timeout, restarted it now.

**Pull request merged**